prov/gni/src/gnix_mr.c

```diff
@@ -440,6 +464,10 @@ int gnix_mr_reg(struct fid *fid, const void *buf, size_t len,
 		}
 	}
 
+	reg_addr = ((uint64_t) buf) & ~((1 << GNIX_MR_PAGE_SHIFT) - 1);
+	reg_len = __calculate_length((uint64_t) buf, len,
+			1 << GNIX_MR_PAGE_SHIFT);
```

Comment:
We found that the length is bogus when using MDD sharing.  You should still be able to read the pagesize out of the MDH then round to find the registration length.